### PR TITLE
feat: #1718 Add a ‘Subscriptions’ table on the ‘Billing’ tab in developers portal

### PR DESCRIPTION
**Pull request by ant066**
# Pull request checklist

## Does this close any currently open issues?

fixes: #1718

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #1718

## What is the new behavior?

Screen Shot 2020-06-16 at 11 19 50 AM
Screen Shot 2020-06-16 at 11 20 04 AM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/marketplace/src/services/subscriptions.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+export type DeleteSubscriptionParams = {
+  id: string
+}
+
+export const fetchSubscriptionsList = async (
+  params: FetchSubscriptionsListParams,
+): Promise<SubscriptionsListResult> => {
+  try {
+    const response = await fetcher({
+      url: `${URLS.subscriptions}?${setQueryParams(params)}`,
```
Comment (tanphamhaiduong) [inline packages/marketplace/src/services/subscriptions.ts:51]: could you use query-string package ?

File: packages/marketplace/src/components/ui/developer-settings/billing/subscriptions.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+  setSubscriptionIdToCancel(id)
+}
+
+export const Subcriptions: React.FC = () => {
+  const dispatch = useDispatch()
+
+  const [isConfirmModalOpen, setIsConfirmModalOpen] = useState(false)
+  const [subscriptionIdToCancel, setSubscriptionIdToCancel] = useState('')
+
+  const handleCloseModal = () => {
```
Comment (tanphamhaiduong) [inline packages/marketplace/src/components/ui/developer-settings/billing/subscriptions.tsx:95]: Pls ensure this function cover by test

File: packages/marketplace/src/sagas/__tests__/developer-subscriptions.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import { takeLatest, put, call, all, fork, select } from 'redux-saga/effects'
+import { Action } from '@/types/core'
+import {
+  FetchSubscriptionsListParams,
+  fetchSubscriptionsList,
+  SubscriptionsListResult,
+  deleteSubscription,
+} from '@/services/subscriptions'
+import ActionTypes from '@/constants/action-types'
+import errorMessages from '../../../../elements/src/utils/validators/error-messages'
```
Comment (willmcvay) [inline packages/marketplace/src/sagas/__tests__/developer-subscriptions.ts:17]: Should be imported from `@reapit/elements`

File: packages/marketplace/src/sagas/developer-subscriptions.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import { logger } from '@reapit/utils'
+import { put, takeLatest, all, fork, call } from 'redux-saga/effects'
+import { errorThrownServer } from '@/actions/error'
+import errorMessages from '../../../elements/src/utils/validators/error-messages'
```
Comment (willmcvay) [inline packages/marketplace/src/sagas/developer-subscriptions.ts:4]: As above, should be from `@reapit/elements`

**Pull request merged**